packages/dds/tree/docs/wip/Modules.md

```diff
@@ -0,0 +1,192 @@
+
+    -   IndirectNode: Single node chunk: most general, supports all functionality (generally want to always include this one)
+    -   UniformSequence: Highly compressed based on tree shape
+
+-   Schema importers: We will likely want at least one tool that takes input schema files and generates the types used by the `Schema` library.
+    Some options are listed here, but little thought has gone into this list:
+
+        -   [json schema](https://json-schema.org/)
+        -   graphQL
+        -   typescript DSL (like [TypeBox](https://www.npmjs.com/package/@sinclair/typebox)) that supports runtime and compile type typing without code gen.
+
+-   Schema libraries: We likely want to provide some useful schema, using and compatible with some importer (above):
+
+    -   json (some way to embed json into our trees)
+    -   xml
+    -   primitives (somehow we need to express primitive types / values. This can be in the form of a schema library, which may or may not be special compared to normal schema libraries)
+
+-   Schematize configurations:
+    Likely the only schematize configuration we will want is one for using it with `Schema` and `EditableTree`, but other options are possible.
+    Likely will want optional fast path for when the `EditableTree` is backed by a `Forest` using shape based compression: may be able to skip some or all validation for whole subtrees this way.
+
+-   Schematize Types: A typescript type-meta-function that subsets the `EditableTree` API to be schema aware (for both reading and editing)
+
+-   Normalizers: Libraries to convert EditableTrees into representations ideal for use in specific frameworks/platforms/tool-kits. Like schematize, may have a fast-path for specific compressed data representations.
+
+    -   react
+    -   custom?
+
+-   Rebaser ChangeSet Representations: There are a lot of different aspects to this
+
+    -   Schema: There are several ways ChangeSets can deal with schema. A ChangeSet may support any combination of the below as additive constraints:
+        -   Support constraints encoded in the edits, which the author of the changeset can use to ensure schema across merges (ex: parent of a node has not changed) (See "Change Tracker" below)
+        -   Support operations that introduce schema into the document as data (collected/tracked by the state tracker), which can be used to validate data (ex: in the change updater)
+        -   Support global/static/hard-coded schema: nothing is needed in the state tracker for these, but they could be applied/enforced just like ones it could track (ex: in the change updater)
+
+-   Rebaser State Trackers: tracks information the rebaser needs to update changes. May also track other information the user might want even if it isn't needed to do the change updating (ex: maybe they want to tree to interpret edits). May need to have their state written into DDS summaries. These are specific to particular ChangeSet implementations.
```

Comment:
Typo: "maybe they want the tree to interpret edits"?
Assuming the above is what you intended, do you mean that they might want access to the tree as additional context within which to interpret edits? If so I think it would help to expand on that, because right now it sounds like the tree is doing the interpretting.